backend/server/src/db/model/drv.rs

```diff
@@ -190,13 +205,80 @@ mod tests {
     }
 }
 
-#[derive(Clone, Debug, FromRow)]
+#[derive(Debug, Clone, FromRow)]
 pub struct Drv {
     /// Derivation identifier.
     pub derivation: DrvId,
 
     /// to reattempt the build (depending on the interruption kind).
     pub system: String,
+
+    pub required_system_features: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct DrvOutput {
+    // nix derivaiton show always structures the output as:
+    // { ${drv}: { ... } }
+    #[serde(flatten)]
+    drvs: HashMap<String, DrvAttrs>,
+}
+
+#[derive(Debug, Deserialize)]
+struct DrvAttrs {
+    pub env: DrvInfo,
+}
+
+#[derive(Debug, Deserialize)]
+struct DrvInfo {
+    /// to reattempt the build (depending on the interruption kind).
+    pub system: String,
+
+    #[serde(rename = "requiredSystemFeatures")]
+    pub required_system_features: Option<String>,
+}
+
+impl Drv {
+    pub fn full_drv_path(&self) -> String {
```

Comment:
Do we need both this and `DrvId::store_path`. I think we can delete this one because you can always do `drv.derivation.store_path()`?

Reply:
Yep, you're right